.github/workflows/build.yml

```
@@ -5,9 +5,9 @@ on: [push, pull_request]
 env:
   COUCH_URL: http://admin:pass@localhost:5984/medic-test
   COUCH_NODE_NAME: nonode@nohost
-  BUILDS_SERVER: _couch/builds_testing
-  STAGING_SERVER: _couch/builds
-  MARKET_URL: ${{ secrets.MARKET_URL }}
+  BUILDS_SERVER: ${{ secrets.AUTH_MARKET_URL && '_couch/builds_testing' || '_couch/builds_external' }}
+  STAGING_SERVER: ${{ secrets.AUTH_MARKET_URL && '_couch/builds' || '_couch/builds_external' }} 
+  MARKET_URL: ${{ secrets.AUTH_MARKET_URL || 'https://staging.dev.medicmobile.org' }}
```

This is so much cleaner! Thank you for sorting this out.

If I understand this change correctly then internal builds will use the URL with the user and pass for everything, which means the URL passed to horti will still contain the URL and still be logged. I think what we need is...

```suggestion
  MARKET_URL_READ: https://staging.dev.medicmobile.org
  MARKET_URL_WRITE: ${{ secrets.AUTH_MARKET_URL || 'https://staging.dev.medicmobile.org' }}
```

This way we only need to use `MARKET_URL_WRITE` in the `node --stack_size=10000 `which grunt` publish-for-testing` and `node ./publish.js` (which will require updates to those scripts to use the new env var name). Everywhere else can use `MARKET_URL_READ`.

Does that make sense?

---

What do you think about the change I made for setting it when the secret is needed in the command. 

EX: `MARKET_URL=MARKET_WRITE_URL node pulish.js`

---

Yes, perfect!

---

It looks like the build is failing on it so there may be a syntax issue, but the idea is good!